### PR TITLE
reconcile cloudstack datacenter

### PR DESCRIPTION
**Pull request by ddjjia**
*Issue #, if available:*
[datacenter reconciler#1267](https://github.com/aws/eks-anywhere-internal/issues/1267)

*Description of changes:*

*Testing (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: controllers/cloudstack_datacenter_controller.go [hunk 1]
```diff
@@ -30,6 +33,42 @@ func (r *CloudStackDatacenterReconciler) SetupWithManager(mgr ctrl.Manager) erro
...
+	}
+
+	patchHelper, err := patch.NewHelper(cloudstackDatacenter, r.client)
+	if err != nil {
+		return ctrl.Result{}, nil
+	}
+
+	defer func() {
+		// Always attempt to patch the object and status after each reconciliation.
+		patchOpts := []patch.Option{}
```
Comment (chrisdoherty4) [inline controllers/cloudstack_datacenter_controller.go:50]: This doesn't seem to be needed?

**Comment by chrisdoherty4** [comment]
I left a few comments before looking at the PR holistically and realizing this seems like boilerplate, is that right?

File: controllers/cloudstack_datacenter_controller.go [hunk 2]
```diff
@@ -30,6 +33,42 @@ func (r *CloudStackDatacenterReconciler) SetupWithManager(mgr ctrl.Manager) erro
...
+
+		if err := patchHelper.Patch(ctx, cloudstackDatacenter, patchOpts...); err != nil {
+			log.Error(reterr, "Failed to patch cloudstackDatacenter")
+			reterr = kerrors.NewAggregate([]error{reterr, err})
+		}
+	}()
+
+	// There's no need to go any further if the cloudstackDatacenter is marked for deletion.
+	if !cloudstackDatacenter.DeletionTimestamp.IsZero() {
```
[the remark this answers is not in the document]
Comment (ddjjia) [reply controllers/cloudstack_datacenter_controller.go:58]: We don't have finalizer currently, which means users can delete those resources, but it's also recoverable. I tried to delete a datacenter and a new one would be created once I tried to scale cp node

**Comment by ddjjia** [comment]
> Is this PR boilerplate only?

Added setdefault for now, will add datacenter validator and validations in later prs

**Review by cxbrowne1207 — COMMENTED** [review]
/lgtm

**Pull request merged**